sources/packages/backend/libs/test-utils/src/factories/application-restriction-bypass.ts

```diff
@@ -0,0 +1,149 @@
+import {
+  Application,
+  ApplicationRestrictionBypass,
+  Note,
+  NoteType,
+  Restriction,
+  RestrictionActionType,
+  RestrictionBypassBehaviors,
+  StudentRestriction,
+  User,
+} from "@sims/sims-db";
+import { E2EDataSources } from "@sims/test-utils/data-source/e2e-data-source";
+import { createFakeNote } from "@sims/test-utils/factories/note";
+import { saveFakeStudentRestriction } from "@sims/test-utils/factories/student-restriction";
+import { ArrayContains, FindOneOptions } from "typeorm";
+
+/**
+ * Create a fake application restrictions bypass.
+ * All required relations entities must be already persisted to database.
+ * @param relations dependencies:
+ * - `application`: related student application.
+ * - `studentRestriction`: active student restriction.
+ * - `creationNote`: note for the bypass creation.
+ * - `bypassCreatedBy`: user that created the bypass.
+ * - `creator`: student user that created the request.
+ * @returns a fake application restrictions bypass.
+ */
+export function createFakeApplicationRestrictionBypass(relations: {
+  application: Application;
+  studentRestriction: StudentRestriction;
+  creationNote: Note;
+  bypassCreatedBy: User;
+  creator?: User;
+}): ApplicationRestrictionBypass {
+  const now = new Date();
+  const bypass = new ApplicationRestrictionBypass();
+  bypass.application = relations.application;
+  bypass.studentRestriction = relations.studentRestriction;
+  bypass.bypassBehavior = RestrictionBypassBehaviors.AllDisbursements;
+  bypass.isActive = false;
+  bypass.creationNote = relations.creationNote;
+  bypass.bypassCreatedBy = relations.bypassCreatedBy;
+  bypass.bypassCreatedDate = now;
+  bypass.removalNote = undefined;
+  bypass.bypassRemovedBy = undefined;
+  bypass.bypassRemovedDate = undefined;
```

Comment:
Is it explicitly required to be set undefined here?

Reply:
Nope, left here to help devs in the future willing to configure the other values.